db/migrate/20200428143500_create_security_policy.rb

```diff
@@ -0,0 +1,55 @@
+class CreateSecurityPolicy < ActiveRecord::Migration[5.0]
+  def change
+    create_table :security_policies do |t|
+      t.string     :type
+      t.string     :name
+      t.string     :description      
+      t.string     :ems_ref
+      t.references :ems,               :type => :bigint
+      t.references :cloud_tenant,      :type => :bigint   
+      t.integer    :sequence_number
+      t.timestamps
+    end
+
+    add_index :security_policies,     :type
+
+    create_table :security_policy_rules do |t|
+      t.string     :type
+      t.string     :name
+      t.string     :description      
+      t.string     :ems_ref
+      t.references :ems,               :type => :bigint
+      t.references :cloud_tenant,      :type => :bigint   
+      t.references :security_policy,   :type => :bigint
+      t.integer    :sequence_number
+      t.string     :status
+      t.string     :action
+      t.string     :direction
+      t.string     :ip_protocol
+      t.timestamps
+    end
+
+    add_index :security_policy_rules, :type
+
+    create_table :security_policy_rule_destination_security_groups, :id => true do |t|
+      t.references :security_policy_rule, :type => :bigint, index: { name: :index_sec_policy_rule_dest_groups_on_security_policy_rule_id }
+      t.references :security_group,       :type => :bigint, index: { name: :index_sec_policy_rule_dest_groups_on_security_group_rule_id }
+    end
+
+    add_index :security_policy_rule_destination_security_groups, [:security_policy_rule_id, :security_group_id], :unique => true, :name => "index_sec_policy_rule_dest_groups"
```

Comment:
@agrare Do we have unique indexes elsewhere in the migrations?  While I'm usually all for database purity, I am concerned about a) the inability to change it during upgrades and b) replication.

Reply:
Yeah we usually do have them for join tables